### PR TITLE
adding support for elastic search in integration framework

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview
This PR includes the change to have Elasticsearch in the integration framework.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- #XXX

## Test Plan

The change was tested manually, and to test the it we are creating an Elasctisearch instance and creating some documents inside an index of that Elasticsearch instance. And creation actionsets for `backup` and `restore` and  that works fine.
To run the tests we can just the env variables that will have aws bucket details and then use `go test` to test the integration framework.

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/testing/integration_test.go [hunk 1]
```diff
@@ -61,6 +61,10 @@ type IntegrationSuite struct {
 	cancel    context.CancelFunc
 }
 
+const (
+	maxRetries = 10
```
Comment (PrasadG193) [inline pkg/testing/integration_test.go]: @viveksinghggits you might want to remove this?

Comment (viveksinghggits) [reply pkg/testing/integration_test.go]: Hi @PrasadG193 ,
Yes, I have removed that.

File: pkg/app/elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	}
+}
+
+func (esi *ElasticsearchInstance) Uninstall(ctx context.Context) error {
+	cli := helm.NewCliClient(helm.V3)
+	log.Print("UnInstalling the application using helm.", field.M{"app": esi.name})
+
+	err := cli.Uninstall(ctx, esi.name, esi.namespace)
+	if err != nil {
+		errors.Wrapf(err, "Error uninstalling the application %s", esi.name)
```
Comment (PrasadG193) [inline pkg/app/elasticsearch.go]: @viveksinghggits you are not storing the value returned by errors.Wrapf(). We can just to `return errors.Wrapf()` here

File: pkg/app/elasticsearch.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	}
+
+	log.Print("Ping to the application was successful.", field.M{"app": esi.name})
+	return nil
+}
+func (esi *ElasticsearchInstance) Insert(ctx context.Context) error {
+	addDocumentToIndexCMD := []string{"sh", "-c", fmt.Sprintf("curl -X POST %s/%s/_doc/?refresh=true -H 'Content-Type: application/json' -d'{\"appname\": \"kanister\" }'", esi.elasticsearchURL, esi.indexname)}
+	_, stderr, err := esi.execCommand(ctx, addDocumentToIndexCMD)
+	if err != nil {
+		errors.Wrapf(err, "Error %s inserting document to an index %s.", stderr, esi.indexname)
```
Comment (PrasadG193) [inline pkg/app/elasticsearch.go]: same here

File: pkg/app/elasticsearch.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+func (esi *ElasticsearchInstance) Init(ctx context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	esi.cli, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
```
Comment (PrasadG193) [inline pkg/app/elasticsearch.go]: we can return err directly

**Pull request merged**